doc/reference/reference_sql/sql_statements_and_clauses.rst

```diff
@@ -496,54 +496,101 @@ Data types may also appear in :ref:`CAST <sql_function_cast>` functions.
+
+    ..  literalinclude:: /code_snippets/test/sql/primary_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_duplicate_author_start
+        :end-before: insert_duplicate_author_end
+        :dedent:
 
-.. code-block:: sql
 
-    -- this is a table with a one-column primary-key constraint
-    CREATE TABLE t1 (s1 INTEGER, PRIMARY KEY (s1));
-    -- this is the column-definition shorthand for the same thing:
-    CREATE TABLE t1 (s1 INTEGER PRIMARY KEY);
-    -- this is a table with a two-column primary-key constraint
-    CREATE TABLE t2 (s1 INTEGER, s2 INTEGER, PRIMARY KEY (s1, s2));
-    -- this is an example of an attempted primary-key violation
-    -- (the third INSERT will fail because 55, 'a' is a duplicate)
-    CREATE TABLE t3 (s1 INTEGER, s2 STRING, PRIMARY KEY (s1, s2));
-    INSERT INTO t3 VALUES (55, 'a');
-    INSERT INTO t3 VALUES (55, 'b');
-    INSERT INTO t3 VALUES (55, 'a');
-
-PRIMARY KEY plus AUTOINCREMENT modifier may be specified in one of two ways: |br|
-- In a column definition after the words PRIMARY KEY, as in ``CREATE TABLE t (c INTEGER PRIMARY KEY AUTOINCREMENT);`` |br|
-- In a PRIMARY KEY (column-list) after a column name, as in ``CREATE TABLE t (c INTEGER, PRIMARY KEY (c AUTOINCREMENT));`` |br|
-When AUTOINCREMENT is specified, the column must be a primary-key column and it must be INTEGER or UNSIGNED. |br|
+**Example 2: two-column primary key**
+
+1.  Create a ``book`` table with the primary key defined on two columns:
+
+    ..  literalinclude:: /code_snippets/test/sql/primary_key_table_constraint_test.lua
+        :language: sql
+        :start-after: create_book_table_start
+        :end-before: create_book_table_end
+        :dedent:
+
+    Insert data into this table:
+
+    ..  literalinclude:: /code_snippets/test/sql/primary_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_books_start
+        :end-before: insert_books_end
+        :dedent:
+
+2.  On an attempt to add the existing book, the following error is raised:
+
+    ..  literalinclude:: /code_snippets/test/sql/primary_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_duplicate_book_start
+        :end-before: insert_duplicate__book_end
+        :dedent:
+
+PRIMARY KEY with the AUTOINCREMENT modifier may be specified in one of two ways:
+
+-   In a column definition after the words PRIMARY KEY, as in ``CREATE TABLE t (c INTEGER PRIMARY KEY AUTOINCREMENT);``
+
+-   In a PRIMARY KEY (column-list) after a column name, as in ``CREATE TABLE t (c INTEGER, PRIMARY KEY (c AUTOINCREMENT));``
+
+When AUTOINCREMENT is specified, the column must be a primary-key column and it must be INTEGER or UNSIGNED.
+
 Only one column in the table may be autoincrement.
 However, it is legal to say ``PRIMARY KEY (a, b, c AUTOINCREMENT)`` -- in that case, there
 are three columns in the primary key but only the first column (``a``) is AUTOINCREMENT.
```

Comment:
I saw this by chance. This is not entirely true. I see that this was added to close https://github.com/tarantool/doc/issues/947, but in fact `AUTOINCREMENT` will be added here to column `c`.

Reply:
Thanks, changed to `... but only the third column (``c``) is AUTOINCREMENT.`